code/datums/components/stationclingy.dm

```diff
@@ -0,0 +1,241 @@
+/// Parent will send a warning message that it's in a space area, because people can slip and fall into other z-levels and lose the item.
+#define IN_SPACE "In Space"
+/// Parent will send a warning message that it's in a mining zone, it's not fond of the climate and animals found in this zone so it'll be unhappy.
+#define IN_MINING "In Mining"
+/// Parent is outside in the rocky wastes of the icemoon, how tragic. It's not happy about that.
+#define IN_ICEMOON "In Icemoon"
+/// Parent will send a message about how we're on a good shuttle going to a valid location! Yippie!
+#define ON_FAVORABLE_SHUTTLE "On Favorable Shuttle"
+/// Parent will send a message about how this shuttle goes somewhere we don't want to go! Uh-oh!
+#define ON_UNFAVORABLE_SHUTTLE "On Unfavorable Shuttle"
+/// Parent will send a message about the nice amenities of the Syndicate Shuttle they are currently on.
+#define ON_SYNDICATE_SHUTTLE "On Syndicate Shuttle"
+/// Parent will send a message about how we're safe inside the station. Just chilling out and hanging around. Good times.
+#define ON_STATION "On Station"
+/// Parent will send a message about how they're thankful that they're safe inside the station. Could also be considered a "Clingy Timer End Message" in how we use it.
+#define BACK_INSIDE_STATION "Back Inside Station"
+/// Parent will send a callous message to anyone who can hear it right before it zoops away.
+#define PISSING_OFF "Pissing Off"
+/// Parent will send a message about how it's going to start counting down until you put it back in a station area.
+#define CLINGY_TIMER_START_MESSAGE "Clingy Timer Start Message"
+
+#define COMPONENT_FORCED_SPEAK_NAME "Clingy Stationloving Component"
+
+///Subtype of stationloving component where the disk is "clingy to the station."
+/datum/component/stationloving/clingy
```

Comment:
talk to ninja about this. we avoid component subtypes in general and the only stuff that does it nowadays is old components CC @ninjanomnom

Reply:
fwiw i think this looks reasonable enough but he has usually been starkly against this

Reply:
i'm okay with reverting it but @tralezab told me to, I know @AnturK also had some thoughts against making it a subtype.